superset/connectors/druid/models.py

```diff
@@ -823,6 +823,7 @@ def granularity(
 
         if origin:
             dttm = utils.parse_human_datetime(origin)
+            assert dttm
```

Comment:
The `parse_human_datetime` allows for an optional `str` and thus returns `None`. This is somewhat of an anti-pattern (the check should occur before the function call) and should be fixed at some stage.

Reply:
Agreed; was going to ask if we could do that change already now, but I expect this to be a Pandora's box, and probably warrants a separate PR to isolate regression risk.